release/src/release-notes.ts

```diff
@@ -118,97 +120,97 @@ enum ProductCategory {
-) => {
-  if (!issueMap[issueType][productCategory]) {
-    issueMap[issueType][productCategory] = [];
-  }
-  issueMap[issueType][productCategory]!.push(issue);
+const getIssueType = (issue: Issue): IssueType => {
+  return match(issue)
+    .when(isNonUserFacingIssue, () => IssueType.underTheHoodIssues)
+    .when(isAlreadyFixedIssue, () => IssueType.alreadyFixedIssues)
+    .when(isBugIssue, () => IssueType.bugFixes)
+    .otherwise(() => IssueType.enhancements);
 };
 
 const getLabels = (issue: Issue): string[] => {
   if (typeof issue.labels === "string") {
-    return [issue.labels];
+    return issue.labels.split(",");
   }
   return issue.labels.map(label => label.name || "");
 };
 
-const hasCategory = (labels: string[], category: string): boolean => {
-  return labels.some(label => label.includes(category));
+const hasCategory = (issue: Issue, categoryName: ProductCategory): boolean => {
+  const labels = getLabels(issue);
+  return labels.some(label => label.includes(categoryName));
 };
 
 export const getProductCategory = (issue: Issue): ProductCategory => {
-  const labels = getLabels(issue);
+  const category = Object.values(ProductCategory).find(categoryName =>
+    hasCategory(issue, categoryName)
+  );
 
-  if (hasCategory(labels, "Administration"))
-    return ProductCategory.administration;
-  if (hasCategory(labels, "Database")) return ProductCategory.database;
-  if (hasCategory(labels, "Embedding")) return ProductCategory.embedding;
-  if (hasCategory(labels, "Operation")) return ProductCategory.operation;
-  if (hasCategory(labels, "Organization")) return ProductCategory.organization;
-  if (hasCategory(labels, "Querying")) return ProductCategory.querying;
-  if (hasCategory(labels, "Reporting")) return ProductCategory.reporting;
-  if (hasCategory(labels, "Visualization"))
-    return ProductCategory.visualization;
-
-  return ProductCategory.other;
+  return category ?? ProductCategory.other;
 };
 
 // Format issues for a single product category
-const formatCategoryIssues = (category: string, issues: Issue[]): string => {
-  return `**${category}**\n\n${issues.map(formatIssue).join("\n")}`;
+const formatIssueCategory = (categoryName: ProductCategory, issues: Issue[]): string => {
+  return `**${categoryName}**\n\n${issues.map(formatIssue).join("\n")}`;
 };
 
 // We want to alphabetize the issues by product category, with "Other" (uncategorized) issues as the caboose
-const sortCategories = (categories: ProductCategory[]): ProductCategory[] => {
+const sortCategories = (categories: ProductCategory[]) => {
   const uncategorizedIssues = categories.filter(
     category => category === ProductCategory.other,
   );
-  return categories
+  const sortedCategories = categories
     .filter(cat => cat !== ProductCategory.other)
-    .sort((a, b) => a.localeCompare(b))
-    .concat(uncategorizedIssues);
+    .sort((a, b) => a.localeCompare(b));
+
+  return [
+    ...sortedCategories,
+    ...uncategorizedIssues,
+  ];
```

Comment:
typescript has trouble with array methods like .concat, this makes it all nice and happy